OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs

```diff
@@ -37,29 +37,44 @@ public static class ConfirmationDialogs
 			var otherButton = prompt.GetOrNull<ButtonWidget>("OTHER_BUTTON");
 
 			var titleMessage = TranslationProvider.GetString(title, titleArguments);
-			prompt.Get<LabelWidget>("PROMPT_TITLE").GetText = () => titleMessage;
+			var titleLabel = prompt.Get<LabelWidget>("PROMPT_TITLE");
+			titleLabel.GetText = () => titleMessage;
 
 			var headerTemplate = prompt.Get<LabelWidget>("PROMPT_TEXT");
 			var textMessage = TranslationProvider.GetString(text, textArguments);
 			var headerLines = textMessage.Split('\n');
 			var headerHeight = 0;
+			var maxWidth = 0;
+			var headerLabels = new List<LabelWidget>(headerLines.Length);
 			foreach (var l in headerLines)
 			{
 				var line = (LabelWidget)headerTemplate.Clone();
 				line.GetText = () => l;
 				line.Bounds.Y += headerHeight;
+
 				prompt.AddChild(line);
+				headerLabels.Add(line);
 
 				headerHeight += headerTemplate.Bounds.Height;
+				maxWidth = Math.Max(maxWidth, line.Bounds.Width / 2);
```

Comment:
Why `/ 2`?

Reply:
My reasoning is that the widget becomes far too wide if it is not halved. This is down to the widget have a base size already. And simply setting the width to `maxWidth` didn't seem to work, unless I messed up in the calculations, which is possible.

Reply:
This doesn't sound right to me.

I'd expect that the yaml template should define the bounds with a reasonable minimum size (such that the buttons don't overlap) and then the code here can only make that larger.

Reply:
Well, that is what is happening. I believe what might have been my issue was I assigning a value below that minimum and the box didn't change shape.

What my code does later in the code is add `maxWidth` to `prompt.Bounds.Width`

Reply:
Here is how big it is without halving the `maxWidth`

![new-too-big](https://github.com/OpenRA/OpenRA/assets/3174205/8bcf9e6b-1ef3-4b74-88bc-86e8d1cb07a9)

Reply:
Hi 👋 

IMO it should be a set amount of buffer distance from the edges. It definitely should not be dynamic
Also do note that this panel is reusable as contains a button in the middle. There needs t be a minimum width as well